### PR TITLE
Create order

### PR DESCRIPTION
**Pull request by RostyslavStepanchuk**

### CODE REVIEW

File: client/src/actions/post.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      })
+      window.location.reload()
+      Toastr.success("Your post was created")
+    })
+}
+
+export const getPostsForHomePage = () => async dispatch => {
+
+  dispatch({
+    type: POSTS_START_LOADING
```
Comment (tarasbashuk) [inline client/src/actions/post.js:53]: For what reason you don't use START/STOP_LOADING constants that we already have?

File: client/src/components/PostFeed/PostFeed.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { getPostsForHomePage, getPostsForProfile } from '../../actions/post'
+import Post from '../Post/Post'
+import Preloader from '../Preloader/Preloader'
+
+class PostFeed extends Component {
```
Comment (tarasbashuk) [inline client/src/components/PostFeed/PostFeed.jsx]: Maybe we should continue to use the function components?

File: client/src/actions/post.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import axios from 'axios'
+import { Toastr } from '../utils/toastr/Toastr'
+import { POSTS_END_LOADING, POSTS_RECIEVED, POSTS_START_LOADING } from '../utils/constants/actionsName'
+
+export const createPost = (uploadForm, setUploadForm) => {
```
Comment (tarasbashuk) [inline client/src/actions/post.js]: I'm not sure that createPost should be at action functions, cause it won't dispatch any action.

File: client/src/reducers/index.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import { combineReducers } from 'redux'
 import { reducer as toastrReducer } from 'react-redux-toastr'
 import auth from './auth'
+import newsFeed from './newsFeed'
 
 export default combineReducers({
   auth,
+  newsFeed,
```
Comment (tarasbashuk) [inline client/src/reducers/index.js]: I would rather call this reducer something like  "posts"

File: src/main/java/com/socialmedia/controller/PostController.java [hunk 5]
```diff
@@ -27,4 +30,10 @@ public PostController(PostMapper postMapper) {
 
     return ResponseEntity.ok(postMapper.getAll());
   }
+
+  @PostMapping
+  public ResponseEntity<PostDtoIn> create(@RequestBody PostDtoIn dtoin) {
+    PostDtoIn input = dtoin;
+    return ResponseEntity.ok(dtoin);
+  }
 }
```
Comment (tarasbashuk) [inline src/main/java/com/socialmedia/controller/PostController.java:39]: I understand that I brake the process, but these changes will lead to conflict at our code.

File: client/src/actions/post.js (hunk 1 above)
Comment (RostyslavStepanchuk) [reply client/src/actions/post.js:53]: Existing constants are for user authentication, they are checked on ProtectedRouter stage. If you stick everything to them, you will see nothing (except spinner) on a page, until news feed is fully downloaded. Though, to my mind, it's better to show basic page (user avatar, some panels, etc) while downloading heavy news feed may take a little longer, so it needs parallel START/LOAD and separate spinner. Also think about infinity scroll in future. You don't want to hide whole page behind spinner while you just getting new bunch of posts from server, do you?

File: client/src/components/PostFeed/PostFeed.jsx (hunk 2 above)
Comment (RostyslavStepanchuk) [reply client/src/components/PostFeed/PostFeed.jsx]: image
I am using component life cycle stages to define it's origin and get posts

File: client/src/reducers/index.js (hunk 4 above)
Comment (RostyslavStepanchuk) [reply client/src/reducers/index.js]: as you wish :) Done

File: src/main/java/com/socialmedia/controller/PostController.java (hunk 5 above)
Comment (RostyslavStepanchuk) [reply src/main/java/com/socialmedia/controller/PostController.java:39]: I needed to have endpoints to check the functionality, also I changed DtoIn. But you can just override these dummies with your own code

File: client/src/components/PostFeed/PostFeed.jsx (hunk 2 above)
Comment (RostyslavStepanchuk) [reply client/src/components/PostFeed/PostFeed.jsx]: Done. Replaced ComponentDidMount with UseEffect

File: client/src/actions/post.js (hunk 3 above)
Comment (RostyslavStepanchuk) [reply client/src/actions/post.js]: Where do you suggest to keep it? I am not a big fan of placing it inside component. Open to discussion, though.

File: client/src/actions/post.js (hunk 1 above)
Comment (tarasbashuk) [reply client/src/actions/post.js:53]: Agree! Next step will be Start/Stop action with own ID ))

File: client/src/actions/post.js (hunk 3 above)
Comment (tarasbashuk) [reply client/src/actions/post.js]: utils/createPost  or something like that

File: client/src/reducers/index.js (hunk 4 above)
Comment (tarasbashuk) [reply client/src/reducers/index.js]: Danke!

File: src/main/java/com/socialmedia/controller/PostController.java (hunk 5 above)
Comment (tarasbashuk) [reply src/main/java/com/socialmedia/controller/PostController.java:39]: OK

File: client/src/components/PostFeed/PostFeed.jsx (hunk 2 above)
Comment (tarasbashuk) [reply client/src/components/PostFeed/PostFeed.jsx]: Thank you!

File: client/src/actions/post.js (hunk 3 above)
Comment (RostyslavStepanchuk) [reply client/src/actions/post.js]: I am against putting it to utils. It's definitely not a helper function.

**Review by skosinskiy — APPROVED** [review]

**Pull request merged**